### ISSUE TITLE
Possibility to export char/varchar data as raw or encoded String

### ISSUE THREAD

**Issue by stoeffu** [darold/ora2pg#1288]
For legacy reasons, I store binary data in varchar fields. In an Oracle database this is not a problem, oracle does not care about the encoding and as long the charactersets are the same on server and client side, no transformation take place. PostgreSQL does not accept binary data in varchar fields, the strings must match the encoding. For that reason I want to store the binary varchar data from Oracle in bytea data in PostgreSQL. I managed to read out the varchar data from oracle as raw (by manipulationg the "set_binmode" function) and the result was promising but not complete because the bytes with the value 0x0 are not exported.

Now my questions:
1. Is there a way to write out also the bytes with the value 0x0?
2. Or even better: Is there a way to write out the varchar values as encoded byte string (E.g. "decode(E'000102..FF', 'hex')")?

Thanks for any suggestions - Stoeffu

**Comment by darold** [comment]
Hi,

You can not stored the nul character in a varchar or text column, you have to use a bytea column.


Have you tried to used MODIFY_TYPE in ora2pg.conf. For example if you want to transform COL3 from table TABLE1 from varchar2 to bytea you could use the following:
```
MODIFY_TYPE     TABLE1:COL3:varcharbytea
```
Is this working for you?

**Comment by stoeffu** [comment]
Hi darold,
The use of MODIFY_TYPE did the trick and I get the output I wanted.
Thank you very much!
I did not find the "varcharbytea" syntax in the documentation.

I mentioned in my original post that I modified the "set_binmode" function to get raw output. The reason for that was the following:
When I set the BINMODE in ora2pg.conf to "iso-8859-1" i get the following error:
`FATAL: can't use open layer :encoding(iso-8859-1)`
When I set the BINMODE in ora2pg.conf to ":raw" i get the following error:
`FATAL: can't use open layer :raw`

The reason why I want raw output ist the fact, that I have char fields which are encoded with the CP850 encoding. This encoding is not supported by PostgreSQL and I have convert the output to UTF-8 before importing to PostgreSQL.

I'm not a perl programmer, but I changed the "set_binmode" function, and it's working for me for both settings but I'm not sure about the "eval" syntax. Tell me if you're interested in my code.
